### ISSUE TITLE
It doesn't work as well as before

### ISSUE THREAD

**Issue by unecc80** [Andre0512/hon#95]
**Describe the bug**
many entities error

**To Reproduce**

**Expected behavior**
It has always worked well in the past. for a little less than a week it has stopped recognizing the entities of the washing machine..

**Screenshots**

![Cattura](https://github.com/Andre0512/hon/assets/139782061/d0896ceb-5475-4d7b-b5a7-e37b2b86aa26)

**Comment by raymondjstone** [comment]
I find the washing machine is working fine for me, the improvements from a month or two ago are huge. I do see one thing (with the last few versions), the error message does not show well. 
![image](https://github.com/Andre0512/hon/assets/9839266/f24afdd4-3da3-4037-8df2-df77c3bf644a)

**Comment by unecc80** [comment]
Not good for me.





Il lun 17 lug 2023, 16:44 Raymond Stone ***@***.***> ha
scritto:

**Comment by unecc** [comment]
maybe candy firmware update can't permit access to sensors and entities?

**Comment by unecc80** [comment]
please help me. 


```
data:
  additional_data:
  appliance:
    PK: user#eu-west-1:3c98fb2c-3e77-4dbb-9aaa-abc6c917d71c
    SK: app#xx-xx-xx-xx-xx-xx
    applianceId: xx-xx-xx-xx-xx-xx#1970-01-01T00:00:00.0Z
    applianceModelId: 1517
    applianceStatus: 1
    applianceTypeId: 1
    applianceTypeName: WM
    attributes:
      acuVersion: 9.A.B
      chipset: ESP32D0WDQ5
      dictionaryId: 308
      fwLabel: iotfw_wmwd
      lang: it-IT
      uiVersion: 3.B.1
    brand: candy
    code: 31018827
    connectivity: wifi|ble
    defaultWarrantyYears: 2
    eepromId: 841
    eepromName: 70012132
    enrollmentDate: 1970-01-01T00:00:00.0Z
    firstEnrollment: True
    firstEnrollmentTBC: True
    fwVersion: 5.13.0
    id: 1517
    lastUpdate: 1970-01-01T00:00:00.0Z
    macAddress: xx-xx-xx-xx-xx-xx
    modelName: RO14126DWMST-S
    nickName: Lavatrice
    purchaseDate: 1970-01-01T00:00:00.0Z
    sections:
      ecoDelayStart: True
      inventory: True
    series: rapido
    topics:
      publish:
      subscribe:
        - $aws/events/presence/disconnected/xx-xx-xx-xx-xx-xx
        - $aws/events/presence/connected/xx-xx-xx-xx-xx-xx
        - haier/things/xx-xx-xx-xx-xx-xx/event/appliancestatus/update
        - haier/things/xx-xx-xx-xx-xx-xx/event/discovery/update
    waterHard: 2
  attributes:
    active: False
    activity:
    commandHistory:
      command:
        ancillaryParameters:
          energyLabel: 4
          liquidDetergentDose: 2
          programCluster: quick
          programFamily: [dashboard]
          programType: W+D
          remoteActionable: 1
          remoteVisible: 1
          suggestedLoadW: 3
          tempContribution: 0
        applianceOptions:
          opt1: prewash
          opt2: hygiene
          opt3: anticrease
          opt4: goodNight
          opt5: extraRinse1
          opt6: extraRinse2
          opt7: extraRinse3
          opt8: acquaplus
        applianceType: WM
        attributes:
          channel: mobileApp
          energyLabel: 4
          origin: standardProgram
          prStr: Rapido 44'
        commandName: startProgram
        device:
          appVersion: 2.1.2
          deviceModel: lahaina
          mobileId: 058feed99b2be60d
          mobileOs: android
          osVersion: 33
        macAddress: xx-xx-xx-xx-xx-xx
        parameters:
          acquaplus: 0
          anticrease: 0
          autoDetergentStatus: 0
          autoSoftenerStatus: 0
          checkUpStatus: 0
          delayTime: 0
          dirtyLevel: 3
          extraRinse1: 0
          extraRinse2: 0
          extraRinse3: 0
          goodNight: 0
          hygiene: 0
          lang: 1
          onOffStatus: 1
          prCode: 11
          prPosition: 7
          prewash: 0
          spinSpeed: 1400
          steamLevel: 0
          temp: 40
        programName: PROGRAMS.WM_WD.RAPID_44_MIN
        timestamp: 1970-01-01T00:00:00.0Z
        transactionId: xx-xx-xx-xx-xx-xx_1970-01-01T00:00:00.0Z
      timestampAccepted: 1970-01-01T00:00:00.0Z
      timestampExecuted: 1970-01-01T00:00:00.0Z
    errors:
      lastConnEvent:
        errorMessage: 1970-01-01T00:00:00.0Z 76943788-3dc8-49e6-9912-4cc7faeec053 Task timed out after 1.00 seconds
    lastConnEvent:
      category: DISCONNECTED
      instantTime: 1970-01-01T00:00:00.0Z
      macAddress: xx-xx-xx-xx-xx-xx
      timestampEvent: 1689623688883
    parameters:
      acquaplus: 0
      anticrease: 0
      apsFreq: 61680
      apsOff: 0
      chartL: 0
      checkUpStatus: 0
      counterMV: 0
      delayTime: 0
      dirtyLevel: 3
      disTestOn: 0
      disTestRes: 0
      dryCotton120: 0
      dryCotton30: 0
      dryCotton60: 0
      dryCotton90: 0
      dryCottonExtra: 0
      dryCottonPA: 0
      dryCottonPS: 0
      dryLevel: 0
      drySynt120: 0
      drySynt30: 0
      drySynt60: 0
      drySynt90: 0
      drySyntExtra: 0
      drySyntPA: 0
      drySyntPS: 0
      dryTime: 0
      error: 00
      extraRinse1: 0
      extraRinse2: 0
      extraRinse3: 0
      fillR: 0
      goodNight: 0
      hygiene: 0
      lang: 0
      machMode: 0
      motS: 0
      nightWashStatus: 0
      ntcD: 7
      ntcW: 152
      numF: 0
      opt9: 0
      prCode: 11
      prCodeFav: 0
      prPhase: 0
      prPosition: 7
      prewash: 0
      program1: 0
      program10: 0
      program11: 0
      program12: 94
      program13: 0
      program14: 0
      program15: 0
      program16: 0
      program17: 0
      program18: 0
      program19: 0
      program2: 1
      program20: 0
      program21: 0
      program3: 2
      program4: 0
      program5: 0
      program6: 0
      program7: 19
      program8: 1
      program9: 5
      remainingTimeMM: 1
      remoteCtrValid: 1
      spinSpeed: 1400
      steamLevel: 0
      t0R: 0
      t0W: 0
      tIW: 0
      temp: 40
      temp0to30: 14
      temp40: 107
      temp60to90: 1
      unbC: 0
      unbF: 0
    pause: False
    programName: No Program
    resultCode: 0
  statistics:
    drumCleaning:
      count: 122
      percentage: 122
      remaining: -22
      tot: 100
    filterCleaning:
      count: 1
      percentage: 1
      remaining: 99
      tot: 100
    lastCheckup: None
    loadingPercentage:
      - current: 2
        date: 1970-01-01T00:00:00.0Z
        max: 2
      - current: 2
        date: 1970-01-01T00:00:00.0Z
        max: 2
      - current: 2.16
        date: 1970-01-01T00:00:00.0Z
        max: 12
      - current: 12
        date: 1970-01-01T00:00:00.0Z
        max: 12
      - current: 3.15
        date: 1970-01-01T00:00:00.0Z
        max: 7
    mostUsedPrograms:
      - count: 48
        prCode: 1
        prPosition: 12
        programName: PROGRAMS.WM_WD.RESISTANT_COTTON
        programType: W+D+S
      - count: 15
        prCode: 11
        prPosition: 7
        programName: PROGRAMS.WM_WD.RAPID_44_MIN
        programType: W+D
      - count: 5
        prCode: 2
        prPosition: 9
        programName: PROGRAMS.WM_WD.DRAIN_SPIN
        programType: W+D
      - count: 2
        prCode: 1
        prPosition: 12
        programName: PROGRAMS.WM_WD.IOT_WASH_BED_LINEN
        programType: W+D+S
      - count: 2
        prCode: 1
        prPosition: 12
        programName: PROGRAMS.WD.RESISTANT_COTTON
        programType: W+D+S
    programsCounter: 76
    resultCode: 0
    temperatureUsage:
      temp0to30: 14
      temp40: 107
      temp60to90: 1
versions:
  hon: 0.9.0
  pyhOn: 0.14.9

**Comment by KiraPC** [comment]
Same error for me @Andre0512 

I've update to new version and I have different KeyErrors

![Screenshot 2023-07-18 alle 15 32 06](https://github.com/Andre0512/hon/assets/46629148/29e0d0c5-5027-4a6e-b81a-a8e949a81653)

And my Washing Dryer is not correctly loaded.
![Screenshot 2023-07-18 alle 15 32 47](https://github.com/Andre0512/hon/assets/46629148/a04504d5-d8c0-4e33-bc85-a96ae7211b74)

**Comment by unecc** [comment]
@KiraPC the same firmware. is it a new update? Maybe it make some changes? 
Waitin for @Andre0512 maybe haier change names of sensors and his plugin does not works for this?

**Comment by Andre0512** [comment]
@unecc, @unecc80 and @KiraPC I will try to reproduce this, please send me the data archive (new button in diagnose) of your devices and the full log message ("load full log" and search for hon)

@raymondjstone thanks! I also got this error message on last machine run and will have a look :slightly_smiling_face:

**Comment by unecc80** [comment]
@Andre0512  please send me your email for send data archive. I've make zip file.

**Comment by Andre0512** [comment]
Upload it here, it's anonymized if you do it like this:
> For further analysis, please add your appliance data archive here (if available)
> Navigate to `Settings` -> `Device & Services` -> `Haier hOn` -> _your device_ and press the _Create Data Archive_ button. 
> Then open notifications to download the data zip archive.
> To attach the file:
> * GitHub Web: Use the "Attach files by dragging & dropping, selecting or pasting them." function
> * GitHub Mobile: Upload the zip archive as image

**Comment by Andre0512** [comment]
Your data looks okay and for me there are all entities available:
![Screenshot from 2023-07-20 22-40-13](https://github.com/Andre0512/hon/assets/20325077/b2ba4a6a-5782-4601-9f62-0824f29621a8)
So this doesn't lead us to a solution. Can you show me which entities exactly aren't available for you? 
And please check the logs and send me the full log message ("load full log" and search for hon)
Did you try to delete the integration and add it again?

**Comment by unecc** [comment]
In order:
- yes, i've try to delete e reinstall hon integration. I've try old
version, new stable version and beta versions. The same problem. the only
possibility to let you see which entities are gone is the old widget. It's
attached.
- i hope  log is correct attached
-  Logger: pyhon.hon

Source: custom_components/hon/__init__.py:30
Integration: Haier hOn (documentation <https://github.com/Andre0512/hon/>,
issues <https://github.com/Andre0512/hon/issues>)
First occurred: 19 luglio 2023 alle ore 09:15:12 (2 occurrences)
Last logged: 19 luglio 2023 alle ore 09:18:21
Device data - {'purchaseDate': '2023-02-07T23:00:00.000Z', 'fwVersion':
'5.13.0', 'waterHard': '2', 'applianceTypeId': 1, 'firstEnrollment': True,
'attributes': {'acuVersion': '9.A.B', 'chipset': 'ESP32D0WDQ5',
'dictionaryId': '308', 'fwLabel': 'iotfw_wmwd', 'lang': 'it-IT',
'uiVersion': '3.B.1'}, 'defaultWarrantyYears': 2, 'applianceModelId': 1517,
'series': 'rapido', 'firstEnrollmentTBC': True, 'code': '31018827', 'SK':
'app#90-38-0c-11-71-10', 'macAddress': '90-38-0c-11-71-10', 'eepromName':
'70012132', 'applianceId': '90-38-0c-11-71-10#2023-02-10T15:01:30Z', 'id':
1517, 'modelName': 'RO14126DWMST-S', 'applianceTypeName': 'WM',
'connectivity': 'wifi|ble', 'serialNumber': '3101882722296204', 'nickName':
'Lavatrice', 'enrollmentDate': '2023-02-10T15:01:30.508Z', 'brand':
'candy', 'lastUpdate': '2023-02-10T15:01:44Z', 'eepromId': 841,
'applianceStatus': 1, 'coords': {'lng': 11.0343388, 'lat': 45.8861938},
'PK': 'user#eu-west-1:3c98fb2c-3e77-4dbb-9aaa-abc6c917d71c', 'sections':
{'inventory': True, 'ecoDelayStart': True}, 'topics': {'publish': [],
'subscribe': ['$aws/events/presence/disconnected/90-38-0c-11-71-10',
'$aws/events/presence/connected/90-38-0c-11-71-10',
'haier/things/90-38-0c-11-71-10/event/appliancestatus/update',
'haier/things/90-38-0c-11-71-10/event/discovery/update']}}

Il giorno gio 20 lug 2023 alle ore 22:51 Andre Basche <
***@***.***> ha scritto:

**Comment by unecc** [comment]
![entities](https://github.com/Andre0512/hon/assets/134554559/795f5753-095e-43e0-8c0f-4a19014f78e5)
[home-assistant_2023-07-22T08-32-22.486Z.log](https://github.com/Andre0512/hon/files/12136029/home-assistant_2023-07-22T08-32-22.486Z.log)

**Comment by Andre0512** [comment]
The log was very helpful, you have a favorite based on `special_39_full_load_steam`, but the program's name is now `iot_special_39_full_load_steam`, so the favorite can't be loaded and the command loader crashed. 
Should be fixed in [`v0.9.1`](https://github.com/Andre0512/hon/releases/tag/v0.9.1) :slightly_smiling_face:

**Comment by unecc** [comment]
Fantastic, now works. Thanks Andre

Il sab 22 lug 2023, 12:13 Andre Basche ***@***.***> ha
scritto:
